### ISSUE TITLE
Improve code quality (JavaScript or Python or Docker)

### ISSUE THREAD

**Issue by niccokunzmann** [fossasia/badgeyay#135]
Currently, we can see [issues with the code](https://www.codacy.com/app/fossasia/badgeyay/issues)

We remove most of them! A clean code improves readability and speeds up development.

Please see the [issues with the code](https://www.codacy.com/app/fossasia/badgeyay/issues) and remove some of them. Create a separate pull-request for each file you change. In some cases, you may want to setup the project to check your changes work.

This is an issue with a lot of small tasks, so you can work in parallel or collaborate as you see fit.

**Comment by jajodiaraghav** [comment]
This is a first-timer issue. Can I still work on this?

**Comment by pushkalkatara** [comment]
Can I work on this?

**Comment by gabru-md** [comment]
@jajodiaraghav wait for sometime if nobody picks this up then you may do it. 
Cheers

**Comment by ivanzaqqa** [comment]
can i still work on this now?

**Comment by invinciblycool** [comment]
I will be interested to take this, if anyone has not been working already.

**Comment by gabru-md** [comment]
@ivanzaqqa and @invinciblycool 
there are more than one files to be corrected.
So you can take one of the files and do the necessary changes and send in a PR.

**Comment by devshiva619** [comment]
I am a first timer and want to contribute to foss, can I get to work on this. This is would be my first ever contribution.

**Comment by gabru-md** [comment]
Sure go ahead!

**Comment by niccokunzmann** [comment]
@devshiva619 @ivanzaqqa @invinciblycool  Hi, you are welcome to contribute. If you have any questions or you like confirmation on how to go on, you can share this here and we can discuss and help. We can help eachother here or on the chat. https://gitter.im/fossasia/badgeyay

**Comment by niccokunzmann** [comment]
This is an issue with a lot of small tasks, so you can work in parallel or collaborate as you see fit.

**Comment by ivanzaqqa** [comment]
@gabru-md i'm sorry what is PR?

**Comment by gabru-md** [comment]
PR means Pull Request @ivanzaqqa

**Comment by niccokunzmann** [comment]
When you have made changes in your repository, you can create a pull-request here: https://github.com/fossasia/badgeyay/pulls -> button "new pull request" and we can see the changes. If you are unsure, we can explain it more in detail. Please ask, then.

**Comment by rupav** [comment]
[https://www.codacy.com/app/fossasia/badgeyay/file/10626798062/issues/source?bid=5114815&fileBranchId=5114815#l13](url) 
![2017-09-30 2](https://user-images.githubusercontent.com/20624380/31046087-149e1796-a610-11e7-990e-15667c7e79aa.png)
@gabru-md @jajodiaraghav  can I work on this?

**Comment by gabru-md** [comment]
Sure go ahead

**Comment by S2606** [comment]
@rupav Yes send a PR.. 👍

**Comment by rupav** [comment]
@S2606 sent the PR :)

**Comment by toonarmycaptain** [comment]
Might be easier to break this into separate issues, but can I work on some of them?

**Comment by devshiva619** [comment]
@toonarmycaptain even I am working on this, so just let me know which file you will be working on so that I don't work on that :smile

**Comment by devshiva619** [comment]
@toonarmycaptain I am working on merge_badges.py

**Comment by toonarmycaptain** [comment]
I found a typo in eventyay_export_anonymizer.py.
Would you like me to pull for each file I edit, or in bulk?

**Comment by devshiva619** [comment]
@toonarmycaptain rectify all the errors in one file and do a PR that's what I do.

**Comment by mohdomama** [comment]
I am interested to work on this. I would like to work on the issue in generate_csv_eventyay.py if nobody is working on it. The `No value for argument 'url_eventyay' in function call tocsv()` can be resolved by passing system arguments in **tocsv()** . That is, if the generate_csv_eventyay.py is run as a standalone program, the `url_eventyay, filename` would have to be passed as system arguments.

**Comment by jajodiaraghav** [comment]
The two file changes after which we can close this is `app/generate-badges.py` and `app/static/js/script.js`.
Cadacy doesn't provide any details regarding how to fix the rest of files

**Comment by devshiva619** [comment]
@jajodiaraghav so should I just work on `app/generate-badges.py` and `app/static/js/script.js` and leave rest of the files?

**Comment by devshiva619** [comment]
@jajodiaraghav ok I will try my best :).
Also I wanted to ask is there anyone else also who has sent PR for `app/merge_badges.py` because in my PR I had changed `os.system` to `subprocess.run` and codacy was not giving any new error and now what I am seeing is that `os.system` has chnged to `subprocess.call` and codacy is giving new errors for the same.

**Comment by jajodiaraghav** [comment]
@djmgit @vaibhavsingh97 Can you please check ^

**Comment by vaibhavsingh97** [comment]
@devshiva619 Please check functionality before sending any PR 😄 
Because of `subprocess.run` badges are not generating. PLease see #269

**Comment by lapmid** [comment]
@niccokunzmann  is it still open

**Comment by devshiva619** [comment]
@lapmid yes its open, I am currently on the issues.

**Comment by lapmid** [comment]
Could i also work on this parallelly or u are assigned to it already

**Comment by devshiva619** [comment]
@lapmid ofcourse you but there's only one file left to be corrected i.e `app/static/css/jquery.minicolors.css` so can start working on this one.

**Comment by lapmid** [comment]
okay m going to work on this

**Comment by lapmid** [comment]
i have done a Pr for that if any other left please tell

**Comment by jajodiaraghav** [comment]
@devshiva619 @lapmid I think making change to the `app/static/css/jquery.minicolors.css` to fix codacy would break the minicolors functionality. The `!important` tag is important there. Do not remove it.

**Comment by gauravchandra7** [comment]
Hi, I am new here, could you tell me any files which nobody has dealt with, I would like to modify that.

**Comment by niccokunzmann** [comment]
Hi, I think, the issues are always updated if there is a modification made.
Which issues do you see? Are they still there?

**Comment by mariobehling** [comment]
This is an ongoing issue and devs should be aware now. Therefore closing it. Cheers.
